tungsten_ci_utils/nexus_cleanup/clean_up.groovy

```diff
@@ -0,0 +1,80 @@
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.common.app.GlobalComponentLookupHelper;
+import org.sonatype.nexus.repository.maintenance.MaintenanceService;
+import org.sonatype.nexus.repository.storage.ComponentMaintenance;
+import org.sonatype.nexus.repository.storage.Query;
+import org.sonatype.nexus.script.plugin.RepositoryApi;
+import org.sonatype.nexus.script.plugin.internal.provisioning.RepositoryApiImpl;
+import com.google.common.collect.ImmutableList;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+
+def retentionDays = 15;
+def tagList = [];
+def repositoryName = 'BartsDockerRepo';
+def whitelisted_tag_suffixes = ["queens", "ocata", "newton", "latest", "40", "94", "122", "129", "161", "214", "309", "360"].toArray();
+log.info(":::Cleanup script started!");
+MaintenanceService service = container.lookup("org.sonatype.nexus.repository.maintenance.MaintenanceService");
+def repo = repository.repositoryManager.get(repositoryName);
+def tx = repo.facet(StorageFacet.class).txSupplier().get();
+def components = null;
+try {
+    tx.begin();
+    components = tx.browseComponents(tx.findBucket(repo));
+} catch (Exception e) {
+    log.info("Error: " + e);
+} finally {
+    if (tx != null)
+        tx.close();
+}
+
+if (components != null) {
+    def retentionDate = DateTime.now().minusDays(retentionDays).dayOfMonth().roundFloorCopy();
+    int deletedComponentCount = 0;
+    def listOfComponents = ImmutableList.copyOf(components);
+
+// comp.version() expected to be in a format {{ os }}-{{ realse }}-{{ branch }}-{{ build }} e.g. rhel-queens-master-386 
+
+    listOfComponents.reverseEach { comp ->
+
+        def tag = comp.version();
+        def tagSplited = tag.split("-");
+        def build_number = tagSplited[tagSplited.length - 1];
+
+        for (j = 0; j < whitelisted_tag_suffixes.length; j++) {
+            if (build_number == whitelisted_tag_suffixes[j]) {
+                log.info("Component skipped: ${comp.name()} ${comp.version()}");
+            } else {
+                if (whitelisted_tag_suffixes.count(build_number) == 0) {
```

Comment:
IMO this is the same as the 'if' above. Same logic. Actually you do not need the `for` loop if there's a `count` method. This statement here will suffice.

Also I think this can be substituted with the list.contains(item) method.